mso/datasource_mso_schema_site_bd.go

```diff
@@ -66,21 +70,23 @@ func dataSourceMSOSchemaSiteBdRead(d *schema.ResourceData, m interface{}) error
 		return fmt.Errorf("No Sites found")
 	}
 	stateSite := d.Get("site_id").(string)
+	stateTemplate := d.Get("template_name").(string)
 	found := false
 	statebd := d.Get("bd_name").(string)
-	for i := 0; i < count; i++ {
+	for i := 0; i < count && !found; i++ {
 		tempCont, err := cont.ArrayElement(i, "sites")
 		if err != nil {
 			return err
 		}
 		apiSite := models.StripQuotes(tempCont.S("siteId").String())
+		apiTemplate := models.StripQuotes(tempCont.S("templateName").String())
 
-		if apiSite == stateSite {
+		if apiSite == stateSite && apiTemplate == stateTemplate {
```

Comment:
Would it be possible to also create to fix in site_bd_subnet for !found and template check? Since you have most of the testing set up already for site_bd and then this would solve https://github.com/CiscoDevNet/terraform-provider-mso/pull/120.

Reply:
yes